### ISSUE TITLE
[Research] Meditation Depth Classification with Random Forest on Consumer EEG (86.7% accuracy)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#45]
## Paper 1
**Title:** EEG-based investigation of effects of mindfulness meditation training on state and trait by deep learning and traditional machine learning
**Authors:** Various
**Year:** 2023 (Frontiers), validated in 2025 clinical trial
**Venue:** Frontiers in Human Neuroscience
**URL:** https://www.frontiersin.org/journals/human-neuroscience/articles/10.3389/fnhum.2023.1033420/full

## Paper 2
**Title:** Consumer-Grade Neurofeedback With Mindfulness Meditation: Meta-Analysis
**Year:** 2025
**Venue:** Journal of Medical Internet Research (JMIR)
**URL:** https://www.jmir.org/2025/1/e68204

## Paper 3
**Title:** Clinical trial: Random Forest meditation classification with EEG (2025)
**Venue:** Frontiers in Human Neuroscience (2025 clinical trial)
**URL:** https://www.frontiersin.org/journals/human-neuroscience/articles/10.3389/fnhum.2025.1616456/pdf

## Key Contribution
A 2025 clinical trial achieved **86.7% classification accuracy, F1=0.84, ROC-AUC=0.89** distinguishing meditation from cognitive tasks using a Random Forest model on theta, alpha, and beta frequency features from consumer-grade EEG. The JMIR meta-analysis (16 studies, 11 using Muse) found that Muse-based neurofeedback produces measurable neural changes, and that device metrics explain ~30% of variance in self-reported trait mindfulness.

Key EEG markers validated across studies:
- **Theta increase** (4-8 Hz) at frontal sites = meditation depth
- **Alpha stabilization** (8-12 Hz) = sustained calm focus
- **Beta suppression** (12-30 Hz) = reduced mental chatter
- **Theta/alpha ratio** = core meditation depth index

## Relevance to NeuralDreamWorkshop
Current `meditation_classifier.py` uses feature-based heuristics with 3 levels (relaxed/meditating/deep). This research provides:
1. A validated Random Forest model architecture that achieves 86.7% on consumer EEG
2. Specific feature sets (theta, alpha, beta power + ratios) validated for meditation classification
3. Evidence that Muse-specific metrics correlate with self-reported mindfulness (r^2 ~ 0.30)
4. Clinical-trial-level validation, not just lab studies

### Files to Modify
- `ml/models/meditation_classifier.py`:
  - Add Random Forest trained model as priority path above feature-based heuristics
  - Feature vector: [theta_power, alpha_power, beta_power, theta_alpha_ratio, alpha_beta_ratio, theta_beta_ratio, spectral_entropy, frontal_midline_theta] per channel
  - Add `compute_meditation_index()` method returning continuous 0-1 score (not just 3 categories)
  - Expand from 3 to 5 depth levels: surface / light / moderate / deep / transcendent
  - Add session-level metrics: % time in each depth, deepest sustained period, average depth
- `ml/training/train_meditation.py` — Create training script:
  - Use existing meditation session data from Muse users
  - Random Forest with 100 trees, max_depth=10
  - Features: per-channel band powers (theta/alpha/beta x 4 channels = 12) + ratios (6) + FMT (1) + spectral_entropy (4) = 23 features
  - 5-fold cross-validation targeting 85%+ accuracy
- `ml/api/routes.py`:
  - Upgrade `/meditation` endpoint to return continuous depth score + session metrics
  - Add `GET /meditation/session-summary` — Post-session depth analysis
  - Add `GET /meditation/trends` — Meditation depth improvement over weeks

### Files to Create
- `ml/models/saved/meditation_rf_model.pkl` — Trained Random Forest model
- `client/src/components/meditation-depth-chart.tsx` — Real-time depth visualization (depth vs time line chart)

### Training Data Strategy
```
Option A: Use Muse's existing meditation recordings (if available from user sessions)
Option B: Collect new data with protocol:
  - 3 min resting baseline (eyes closed)
  - 10 min guided meditation (audio prompt)
  - 3 min cognitive task (mental arithmetic)
  - Label: resting=0, meditation=1, task=2
  - Collect from 20+ participants
Option C: Use public datasets (EEG meditation datasets on PhysioNet)
```

### Meditation Traditions Mapping (from CLAUDE.md)
```
Focused attention (Shamatha): high alpha, moderate theta → "moderate" depth
Open monitoring (Vipassana): high theta, wide alpha → "deep" depth
Non-dual (Dzogchen): gamma bursts + theta-gamma coupling → "transcendent" depth
Loving-kindness (Metta): alpha + gamma, high coherence → "deep" depth
```

## Expected Impact
- **Accuracy:** 86.7% meditation vs non-meditation (up from heuristic-based)
- **Depth classification:** 75-80% across 5 depth levels
- **Clinical validation:** Based on clinical trial data, not just lab studies
- **User value:** Quantified meditation progress over weeks/months

## Complexity: LOW-MEDIUM
- Random Forest is already in scikit-learn (no new dependencies)
- Feature extraction reuses existing `extract_band_powers()` + `compute_frontal_midline_theta()`
- Main effort: collecting/curating meditation training data
- UI updates are incremental (add depth chart to existing meditation page)
